File: baekjoon/p01065/Main.java
package baekjoon.p01065;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println(solve(sc.nextInt()));
    }

    private static int solve(int n) {
        int ret = 0;
        for (int i = 1; i <= n; i++) {
            if (check(i)) {
                ret++;
            }
        }

        return ret;
    }

    private static boolean check(int n) {
        if (n < 100) {
            return true;
        }

        String str = String.valueOf(n);
        int[] arr = new int[str.length()];
        for (int i = 0; i < arr.length; i++) {
            arr[i] = str.charAt(i) - '0';
        }

        int diff = arr[0] - arr[1];
        for (int i = 0; i < arr.length - 1; i++) {
            if (arr[i] - arr[i + 1] != diff) {
                return false;
            }
        }

        return true;
    }

}

File: baekjoon/p01065/Main.kt
package baekjoon.p01065

fun main() {
    println(solve(readLine()!!.toInt()))
}

private fun solve(n: Int): Int {
    return (1..n).count { check(it) }
}

private fun check(n: Int): Boolean {
    if (n < 100) {
        return true
    }

    val arr = n.toString().map { it - '0' }
    val diff = arr[0] - arr[1]

    for (i in 0 until arr.size - 1) {
        if (arr[i] - arr[i + 1] != diff) {
            return false
        }
    }

    return true
}